src/react/hooks/useBackgroundQuery.ts

```diff
@@ -230,22 +235,13 @@ export function useBackgroundQuery<
     (variables) => {
       const promise = queryRef.refetch(variables);
 
-      setPromiseCache((promiseCache) =>
-        new Map(promiseCache).set(queryRef.key, queryRef.promise)
-      );
+      setWrappedQueryRef({ current: wrapQueryRef(queryRef, queryRef.promise) });
 
       return promise;
     },
     [queryRef]
   );
 
-  queryRef.promiseCache = promiseCache;
-
-  const wrappedQueryRef = React.useMemo(
-    () => wrapQueryRef(queryRef),
-    [queryRef]
-  );
-
   return [
     didFetchResult.current ? wrappedQueryRef : void 0,
```

Comment:
Continuing previous suggestion, perhaps you can just use `wrapped.current` here?

```suggestion
    didFetchResult.current ? wrapped.current : void 0,
```

Reply:
I believe this comment might be obsolete with the changes I just pushed.